tools/nnyeah/nnyeah/Program.cs

```diff
@@ -1,23 +1,84 @@
 using System;
 using System.IO;
+using System.Reflection;
+using Mono.Options;
+using System.Collections.Generic;
 
 #nullable enable
 
 namespace nnyeah {
-    class Program {
-        static void Main (string [] args)
-        {
-            if (args.Length != 2) {
-                Console.Error.WriteLine ("Usage: dotnet nnyeah.dll /path/to/input/file.dll /path/to/output/file.dll");
-                Environment.Exit (1);
-            }
-            using var stm = new FileStream (args [0], FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
-            var reworker = new Reworker (stm);
-            reworker.Load ();
-            if (reworker.NeedsReworking ()) {
-                using var ostm = new FileStream (args [1], FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite);
-                reworker.Rework (ostm);
-            }
-        }
-    }
-}    
+	class Program {
+		static void Main (string [] args)
+		{
+			var doHelp = false;
+			string? infile = null, outfile = null;
+			var verbose = false;
+			var suppressWarnings = false;
+			var warnings = new List<string> ();
+			var transforms = new List<string> ();
+
+			var options = new OptionSet () {
+				{ "h|?|help", o => doHelp = true },
+				{ "i=|input=", f => infile = f },
+				{ "o=|output=", f => outfile = f },
+				{ "v|verbose", o => verbose = true },
+				{ "s|suppress-warnings", o => suppressWarnings = true },
+			};
+
+			try {
+				var extra = options.Parse (args);
+			} catch {
+				doHelp = true;
+			}
+
+
+			if (doHelp || infile is null || outfile is null) {
+				PrintOptions (options, Console.Out);
+				Environment.Exit (0);
+			}
+
+			if (!File.Exists (infile)) {
+				Console.Error.WriteLine ($"input file '{infile}' doesn't exist.");
+				Environment.Exit (1);
+			}
+
+			using var stm = new FileStream (infile, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+			var reworker = new Reworker (stm);
+
+			try {
+				reworker.Load ();
+			} catch (Exception e) {
+				Console.Error.WriteLine ($"Unable to read module from file {infile}: {e.Message}.");
+				Environment.Exit (1);
+			}
+
+			reworker.WarningIssued += (s, e) => warnings.Add (e.HelpfulMessage ());
+			reworker.Transformed += (s, e) => warnings.Add (e.HelpfulMessage ());
+
+			if (reworker.NeedsReworking ()) {
+				try {
+					using var ostm = new FileStream (outfile, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite);
```

Comment:
This FileStream is going to be problematic if the file is already present (user wanted to override or simply was wrong). This will override the file no matter what. We should throw an exception if that was not intended and add a -f or --force to the arguments. We can add the new argument in a diff PR, but we should already check for the existence in this one.